src/core/account/account.ts

```diff
@@ -0,0 +1,139 @@
+import type { AccountAuthenticator } from "../../transactions/authenticator/account";
+import { HexInput, SigningSchemeInput } from "../../types";
+import type { AccountAddress, AccountAddressInput } from "../accountAddress";
+import { Ed25519PrivateKey } from "../crypto";
+import { Ed25519Account } from "./ed25519";
+import { type PrivateKeyInput, SingleKeyAccount } from "./singleKey";
+
+export interface CreateEd25519AccountFromPrivateKeyArgs {
+  privateKey: Ed25519PrivateKey;
+  address?: AccountAddressInput;
+  legacy?: true;
+}
+
+export interface CreateEd25519SingleKeyAccountFromPrivateKeyArgs {
```

Comment:
I think it is worth to put up some explanation here on each type. As Single Key is not well known devs might be confusing on the difference between `CreateEd25519Account` , `CreateEd25519SingleKeyAccount`, `CreateSingleKeyAccount` etc